src/ocamlorg_frontend/dune

```diff
@@ -1,6 +1,12 @@
 (library
  (name ocamlorg_frontend)
- (libraries ocamlorg.global dream ood timedesc ocamlorg_static))
+ (libraries
+  ocamlorg.global
+  dream
+  ood
+  timedesc
+  ocamlorg_static
+  ocamlorg_package))
```

Comment:
Ah yes! That was a design principle used when creating the website: the front end should be self-contained and not depend on anything, ensuring that it remains only a small wrapper around HTML and that all the logic lives elsewhere.

Reply:
Thanks, @sabine for pointing this out. Indeed, there's a situation here.

Here is a diagram of the internal dependencies inside ocamlorg.

https://drive.google.com/file/d/1_kBOQx-0naKhGcVc0JL1kdevL0fA8_IW/view?usp=sharing

Drawing an arrow from `package` to `frontend` isn't great. I did it to provide to the `frontend` the build check status (from `check.ci.ocaml.org`) as a datatype rather than a `string`, which would have put logic in `frontend`, which would have been unquestionably wrong. 

I don't think using `package.ml` from the `frontend` is a great solution either. If I was to put the definition of `Status.t` that is currently in `ocamlorg_package/lib/build.ml` there, it would revert the offending arrow. 

Therefore, I see two ways to go (but there are certainly more):

1. Use a `Stdlib` predefined type to encode the build check status `Build.Status.t`. For instance, the JSON coming from `check.ci.ocaml.org` could be parsed such that the statuses would have type `string option` where `None` would mean success and `Some cause` would mean failure.
1. Move that build check status type up to either `ood`, `global` or something else which would lie at the side of those in the diagram.

To me, the second one makes more sense. The first one feels bizarre. Both `package` and `frontend` already rely on types defined in `odd`. Data coming from `check.ci.ocaml.org` is yet another kind of data that needs to be dealt with. The only difference is encoding: JSON instead of Yaml.

What's your view on this?

Reply:
`Ocamlorg_frontend.Package` provides only rendering-related logic and types. So we indeed cannot move the definition of `Status.t` - as it currently is - in there. The current definition of `Status.t` contains both data import logic and rendering logic. So that would have to be split up into these two parts.

(2) sounds fine to me because this is indeed just another data import from an external source.

I think we should disentangle the different data sources, in general.

As I see it, `Ocamlorg_package` is on the same "level" as `ood`: it's a module that provides data to ocamlorg web. However, among all the data modules, it's the most convoluted one because it imports from several data sources: (1) from opam repository, (2) from the documentation pipeline.

In a traditional Model-View-Controller architecture, `ood`, `package` would be models, while `web` is the controller, and `frontend` is the view.

The special situation about `ood` is that there's no need for `frontend` to repeat the types in `ood` because they're already in the shape that we need to render a template. The architecturally clean thing to do here would be to only share the types of `ood` with the frontend, but not the values - the controller is who determines which values get rendered.

Reply:
Tiny addendum: the critical thing not to share with the frontend IMO is all the data import logic